cli/cli-improvements.md

```diff
@@ -0,0 +1,315 @@
+*	Workspace – Radius workspace has a "per user" information about the kubernetes context/ cluster where the Radius services are hosted as well as the “default” environment and resource group  of the user.
+
+There are a few points that can be improved in the way these concepts work together:
+
+*	Today, when we rad init,  we make a “default” resource group which creates a single resource group perception for radius users. However, we would want multiple resource groups for multiple copies of same application, based on the scenario. For instance, a cool-app deployment to production and the same cool-app deployment to staging would not need the same RBAC and hence have to be two copies in two resource groups. Also, they have to be in different resource groups so that there are no resource-id conflicts. Would it be a better idea, to not create default group during rad init, and create/use a resource group based on what the users chooses as the application name and its environment [production, staging,[enter your own]]?
+
+
+*	Today, if we want two copies of same app, we want to create two resource group, have an environment in each of those resource group, have the user explicitly use rad group switch and rad env switch or specify the group and env using flags, to make a desirable deployment. Would it be a better idea to, instead of expecting users specify instructions explicitly, print a default behavior which is intuitive and deploy the app if user confirms?
+
+
+*	Since only ResourceGroup  is part of resource ID, if we deploy multiple application to same resource group (ex: we do this in our functional tests) we have to ensure the resource names across applications are all unique. This limitation has no reason to be present in real world and is a hinderance since the developer has to make sure to choose names that are not already taken across applications. Again, we could create a resource group based on user scenario per instance of application using an agreed upon convention to solve this problem. 
+
+## Terms and definitions
+
+| Term | Definition |
+|---|---|
+| workspace | |
+| environment | |
+| scope | |
+
+
+## Objectives
+
+### Goals
+
+Enhance the user experience with `rad deploy` by making its interaction with various Radius concepts more intuitive.
+
+### Non-Goals
+
+NA
+
+### User scenarios
+
+* As a developer, I would prefer not to manually enter or set the group to which my application will be deployed each time. It would be beneficial if Radius could automatically select the group name, ensuring all related resources are grouped together and avoiding conflicts with resources from other applications. I should have the option to override this behavior when necessary.
+
+* As a developer, I would like to avoid the responsibility of selecting unique names for components within my application to prevent conflicts with other deployed applications.
+
+## Design
+
```

Comment:
This is a good place to explain the current behavior. Suggestion

- What does `rad` do today with respect to resource groups? (and what's the problem with it)
- What would happen if we took out the idea of "default scope" from `config.yaml`
- Once we do that, many commands become very verbose, how would we simplify things.